control_plane/src/endpoint.rs

```diff
@@ -1021,8 +978,18 @@ impl Endpoint {
         }
     }
 
-    pub fn stop(&self, mode: &str, destroy: bool) -> Result<()> {
-        self.pg_ctl(&["-m", mode, "stop"], &None)?;
+    pub async fn stop(&self, mode: &str, destroy: bool) -> Result<TerminateResponse> {
+        let ip = self.external_http_address.ip();
+        let port = self.external_http_address.port();
+        // We ignore _mode_ and always pass "immediate", otherwise /terminate will
+        // wait 30s before returning which breaks tests
+        let url = format!("http://{ip}:{port}/terminate?mode=immediate");
+        let token = self.generate_jwt(Some(ComputeClaimsScope::Admin))?;
+        let request = reqwest::Client::new().post(url).bearer_auth(token);
+        let response = request.send().await.context("/terminate")?;
+        let text = response.text().await.context("/terminate result")?;
+        let response: TerminateResponse =
+            serde_json::from_str(&text).with_context(|| format!("deserializing {text}"))?;
```

Comment:
Do you mind changing `mode: &str` to the enum `TerminateMode`? And then, implementing "Display"  if it's useful.

Reply:
Fixed